### PR TITLE
Adding MorseCode and Projectile files including tests

### PR DESCRIPTION
**Pull request by emmyrunge**

### CODE REVIEW

File: src/runge/projectile/Projectile.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+public class Projectile
+{
+    private final double angle;
+    private final double velocity;
+
+    private final double radians;
+    private final double time;
+
+    public Projectile(double angle, double velocity, double time)
```
Comment (andrewoid) [inline src/runge/projectile/Projectile.java]: Time starts at 0. It should not be passed in.

File: src/runge/projectile/Projectile.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    {
+        return Math.sin(radians) * velocity * time - .5 * 9.8 * Math.pow(time, 2);
+
+    }
+
+    public double incTime(double delta)
+    {
+
+        return time*delta;
+    }
```
Comment (andrewoid) [inline src/runge/projectile/Projectile.java]: This is wrong.

File: src/runge/projectile/Projectile.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    public double getX(double time)
+    {
+        return Math.cos(radians) * velocity * time;
+    }
+
+    public double getY(double time)
+    {
+        return Math.sin(radians) * velocity * time - .5 * 9.8 * Math.pow(time, 2);
```
Comment (andrewoid) [inline src/runge/projectile/Projectile.java]: time * time

File: src/runge/projectile/Projectile.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    }
+}
+
+
+/*
+for(int time = 0; time <= 10; time++)
+        {
+            System.out.println("x is: " + projectile.getX(time) + ", " + "y is: " + projectile.getY(time));
+        }
+ */
```
Comment (andrewoid) [inline src/runge/projectile/Projectile.java]: You don't need this commented out code.

File: src/runge/projectile/Projectile.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    public double getVelocity()
+    {
+        return velocity;
+    }
+    public double getTime()
+    {
+        return time;
+    }
+
+    public double getX(double time)
```
Comment (andrewoid) [inline src/runge/projectile/Projectile.java]: If time exists in the projectile now, why are you passing it in?

File: src/runge/projectile/MorseCode.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
+package runge.projectile;
+
+import java.util.Arrays;
+
+public class MorseCode
+{
+    private final String[] morse;
+    String phrase;
```
Comment (andrewoid) [inline src/runge/projectile/MorseCode.java]: What is "phrase"?

File: src/runge/projectile/MorseCodeTest.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+                's', 't', 'u', 'v', 'w', 'x', 'y', 'z'};
+
+        //given
+        MorseCode morseCode = new MorseCode(morse, english);
+
+        //when
+        String phrase = "is this gonna work?";
+
+        //then
+        morseCode.getPhrase(phrase);
```
Comment (andrewoid) [inline src/runge/projectile/MorseCodeTest.java]: You need to assertEquals() that the result of calling this method is what you expect it to be.

That's how you know if it works.

File: src/runge/projectile/Projectile.java (hunk 2 above)
Comment (emmyrunge) [reply src/runge/projectile/Projectile.java]: you said that time is incremented by delta, and that's what I'm doing here.

File: src/runge/projectile/Projectile.java (hunk 1 above)
Comment (andrewoid) [reply src/runge/projectile/Projectile.java:11]: This method should not return a value.

And this code is not incrementing time.

File: src/runge/projectile/Projectile.java (hunk 3 above)
Comment (emmyrunge) [reply src/runge/projectile/Projectile.java]: isn't the equation velocity * time? if that's not where you are referring to then where?

File: src/runge/projectile/Projectile.java (hunk 2 above)
Comment (andrewoid) [reply src/runge/projectile/Projectile.java]: This method should not return a value.

And this code is not incrementing time.

File: src/runge/projectile/Projectile.java (hunk 3 above)
Comment (andrewoid) [reply src/runge/projectile/Projectile.java]: Do time * time instead of using pow

File: src/runge/projectile/Main.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+public class Main
+{
+    public static void main(String [] args)
+    {
+        Projectile projectile = new Projectile(39, 77);
+
+        System.out.println("x is: " + projectile.getX() + ", " + "y is: " + projectile.getY());
+
+        projectile.incTime(2);
```
Comment (andrewoid) [inline src/runge/projectile/Main.java]: Why are you increasing time here?

File: src/runge/projectile/Main.java [hunk 9]
```diff
@@ -0,0 +1,15 @@
+package runge.projectile;
+
+//import runge.projectile.Projectile;
```
Comment (andrewoid) [inline src/runge/projectile/Main.java]: Remove commented out code

File: src/runge/projectile/MorseCode.java [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+    private final String[] morse;
+    private final char[] english;
+
+    public MorseCode(String[] morse, char[] english)
+    {
+        this.morse = morse;
+        this.english = english;
+    }
+
+    public StringBuilder getPhrase(String phrase)
```
Comment (andrewoid) [inline src/runge/projectile/MorseCode.java]: Return a String. Not a StringBuilder

Comment (andrewoid) [reply src/runge/projectile/MorseCode.java]: The method was supposed to be called toMorseCode

File: src/runge/projectile/Projectile.java [hunk 11]
```diff
@@ -0,0 +1,48 @@
+package runge.projectile;
+
+public class Projectile
+{
+    private final double angle;
+    private final double velocity;
+
+    private final double radians;
+    private double time = 6;
```
Comment (andrewoid) [inline src/runge/projectile/Projectile.java]: Why are you starting this off at 6?

File: src/runge/projectile/ProjectileTest.java [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+    //need to test to incTime
+    @Test
+    public void incTime()
+    {
+        //given
+        Projectile projectile = new Projectile(39, 77);
+
+        //when
+        projectile.incTime(.0001);
+        int newTime = (int) projectile.getTime();
```
Comment (andrewoid) [inline src/runge/projectile/ProjectileTest.java]: Why are you casting this?

File: src/runge/projectile/ProjectileTest.java [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+
+        //then
+        assertEquals(6, newTime);
```
Comment (andrewoid) [inline src/runge/projectile/ProjectileTest.java]: This obviously shouldn't be right.

**Review by andrewoid — CHANGES_REQUESTED** [review]

File: src/runge/projectile/MorseCode.java [hunk 14]
```diff
@@ -37,7 +35,42 @@ public StringBuilder getPhrase(String phrase)
                 }
             }
         }
-        return morseSentence;
+        return String.valueOf(morseSentence);
```
Comment (andrewoid) [inline src/runge/projectile/MorseCode.java]: `morseSentence.toString()` instead

File: src/runge/projectile/MorseCode.java [hunk 15]
```diff
@@ -37,7 +35,42 @@ public StringBuilder getPhrase(String phrase)
...
+                        if (morseCodeLetter[j].equals(morse[k]))
+                        {
+                            englishSentence.append(english[k]);
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+        return String.valueOf(englishSentence);
```
Comment (andrewoid) [inline src/runge/projectile/MorseCode.java]: `englishSentence.toString()`

File: src/runge/projectile/MorseCodeTest.java [hunk 16]
```diff
@@ -25,9 +25,33 @@ public void getPhrase()
         MorseCode morseCode = new MorseCode(morse, english);
 
         //when
-        String phrase = String.valueOf(morseCode.getPhrase("hey"));
+        String phrase = String.valueOf(morseCode.toMorseCode("hey"));
```
Comment (andrewoid) [inline src/runge/projectile/MorseCodeTest.java]: `String.valueOf()` is unnecessary

File: src/runge/projectile/Projectile.java [hunk 17]
```diff
@@ -37,12 +38,11 @@ public double getX()
     public double getY()
     {
         return Math.sin(radians) * velocity * time - .5 * 9.8 * (time*time);
-
     }
 
     public void incTime(double delta)
     {
-        this.time += delta;
+        this.time = this.time + delta;
```
Comment (andrewoid) [inline src/runge/projectile/Projectile.java]: using `+=` was better

File: src/runge/projectile/MorseCode.java [hunk 18]
```diff
@@ -37,7 +35,42 @@ public StringBuilder getPhrase(String phrase)
...
+        StringBuilder englishSentence = new StringBuilder();
+
+        String [] morseCodeWord = morseCode.split("/");
+
+        String [] morseCodeLetter = morseCode.split(" , ");
+
+
+        //break down sentence to word
+        //break down word to letter
+        for (int i = 0; i < morseCodeWord.length; i++)
```
Comment (andrewoid) [inline src/runge/projectile/MorseCode.java]: What is `i` used for?

File: src/runge/projectile/MorseCodeTest.java [hunk 19]
```diff
@@ -25,9 +25,33 @@ public void getPhrase()
...
 
         //then
         Assertions.assertEquals(" .... , . , -.-- , ", phrase);
     }
+
+    @Test
+    public void toEnglish()
```
Comment (andrewoid) [inline src/runge/projectile/MorseCodeTest.java:36]: Try writing a test with multiple words.

**Comment by emmyrunge** [comment]
I’ve been working on the last comment for a few hours now and can’t seem to
fix it. do you have time tonight to help?

**Comment by andrewoid** [comment]
Post your questions as a reply to my comment in PR instead of emailing me.

We should be able to meet tonight. What time?

**Comment by emmyrunge** [comment]
Will do. Does 9:30 work?

**Comment by andrewoid** [comment]
Yes.

**Comment by andrewoid** [comment]
https://zoom.us/s/4616736274

**Review by andrewoid — APPROVED** [review]

**Pull request merged**